### PR TITLE
Tweak lints and checks for tools/quic

### PR DESCRIPTION
**Pull request by Hexcles**
in preparation for #22844

Tracking issue: #19114

### CODE REVIEW

**Comment by Hexcles** [comment]
I promise I'll add proper support to manifest later, as per the RFC.

Also we should probably run `mypy` in Python 3 mode against `tools/quic`, but that's also for a later day.

File: tools/lint/lint.py [hunk 1]
```diff
@@ -665,6 +666,8 @@ def check_script_metadata(repo_root, path, f):
                 pass
             elif key == b"variant":
                 pass
+            elif key == b"quic":
```
Comment (jgraham) [inline tools/lint/lint.py:669]: I feel like at some point this should be 
```
elif key not in (b"title", b"script", b"variant", b"quic"):
     errors.append(rules.UnknownMetadata.error(path, line_no=idx + 1))
```

**Review by jgraham — APPROVED** [review]

File: tools/lint/lint.py (hunk 1 above)
Comment (Hexcles) [reply tools/lint/lint.py:669]: Let's make today the day. :)

File: tools/lint/lint.py [hunk 2]
```diff
@@ -586,7 +586,8 @@ def check(self, root):
 
 def check_python_ast(repo_root, path, f):
     # type: (str, str, IO[bytes]) -> List[rules.Error]
-    if not path.endswith(".py"):
+    # *.quic.py are Python 3 only and cannot be parsed by Python 2.
+    if (not path.endswith(".py")) or path.endswith(".quic.py"):
```
[the remark this answers is not in the document]
Comment (Hexcles) [reply tools/lint/lint.py:590]: I'll leave this one for another day...

**Pull request merged**